OpenRA.Game/Map/Map.cs

```diff
@@ -356,14 +356,14 @@ public Map(ModData modData, IReadOnlyPackage package)
 					{
 						for (var j = 0; j < MapSize.Y; j++)
 						{
-							var tile = s.ReadUInt16();
+							var type = s.ReadUInt16();
 							var index = s.ReadUInt8();
 
 							// TODO: Remember to remove this when rewriting tile variants / PickAny
 							if (index == byte.MaxValue)
 								index = (byte)(i % 4 + (j % 4) * 4);
 
-							Tiles[new MPos(i, j)] = new TerrainTile(tile, index);
+							Tiles[new MPos(i, j)] = new TerrainTile(type, index);
```

Comment:
This seems to be heading into the wrong direction. The tile is the :black_square_button: itself, the type would be `Gras`/`Road` etc.
https://github.com/OpenRA/OpenRA/blob/442d91537ee2f104d844738476b0caaa9ed5c664/OpenRA.Mods.Common/Scripting/Global/MapGlobal.cs#L93-L94

Reply:
> This seems to be heading into the wrong direction. The tile is the 🔲 itself, the type would be `Gras`/`Road` etc.
> https://github.com/OpenRA/OpenRA/blob/442d91537ee2f104d844738476b0caaa9ed5c664/OpenRA.Mods.Common/Scripting/Global/MapGlobal.cs#L93-L94

Yes but a TerrainTile is defined by its type and its index, as can be seen in the TileReference.cs file. And if a conversion needs to be made from tile -> type, this should be explictly stated, however no such conversion is occurring here. If you really want to be explicit you could call it tileType, but then TerrainTile should be modified to take a tileType parameter, which I don't think is ideal. Pchote mentioned these "type" and "index" parameters are a legacy naming convention from Westwood, so they probably shouldn't be renamed unless there is a good reason to.

![image](https://user-images.githubusercontent.com/46154431/131345056-53578e55-4001-401f-825c-6f5a570fb17b.png)

Reply:
Understand, however I have found that the change I am proposing above already exists 5 lines below with ResourceTile, as shown below. Therefore we need to choose one approach or the other, as currently both approaches are being used ("tile" is passed in to TerrainTile, while "type" is being passed in to ResourceTile). My position on this was already mentioned, but am keen to hear what others think, as I think being inconsistent the way it is now is the worst possible outcome (and it doesn't help that there are no comments articulating any of this complexity).

https://github.com/OpenRA/OpenRA/blob/442d91537ee2f104d844738476b0caaa9ed5c664/OpenRA.Game/Map/Map.cs#L371-L383

Reply:
Resources are not terrain tiles. They are sprite overlays. I think you should try and get familiar with the code base before suggesting large scale internal API renamings no one has asked for.

Reply:
Resources are not terrain tiles, but sprite overlays defined by a trait. It is not a good example for consistent naming.

I think you should have a look at the ` .yaml` definition. What you call a type is actually called ID there https://github.com/OpenRA/OpenRA/blob/442d91537ee2f104d844738476b0caaa9ed5c664/mods/ra/tilesets/temperat.yaml#L116-L122

and yes it is actually redundant. https://github.com/OpenRA/OpenRA/issues/3222 

Note that type already has a different meaning:

https://github.com/OpenRA/OpenRA/blob/442d91537ee2f104d844738476b0caaa9ed5c664/mods/ra/tilesets/temperat.yaml#L63-L67

Reply:
Ok, though I still don't think a var called 'tile' should be passed into a parameter called 'type', I have reverted this particular change to what it was originally.